pandas/core/reshape/reshape.py

```diff
@@ -1045,7 +1049,9 @@ def get_empty_frame(data) -> DataFrame:
         return DataFrame(dummy_mat, index=index, columns=dummy_cols)
 
 
-def _reorder_for_extension_array_stack(arr, n_rows: int, n_columns: int):
+def _reorder_for_extension_array_stack(
+    arr: ExtensionArray, n_rows: int, n_columns: int
+) -> ExtensionArray:
```

Comment:
probably should be a typevar. but also ok to leave as is for now if not needed as this is internal